### ISSUE TITLE
refactor(ledger-connector): implement a base type for both endpoints, with or without keychain

### ISSUE THREAD

**Issue by elenaizaguirre** [hyperledger/cactus#1658]
When we implement endpoints without the keychain dependency in the different connectors, we are adding optional parameters in the request objects that are not needed in the request itself but are needed in calls to internal common methods of the connectors.
We will define these optional parameters (together with those necessary to call the ledger) in a base type (InvokeRequestBaseV1 and DeployRequestBaseV1) that will be used by both endpoints with or without keychain internally, as explained in [this conversation](https://github.com/hyperledger/cactus/pull/1615#discussion_r766970762).
This way we will use the new object in the internal calls of the connector without the need to add parameters in the request object that are not necessary.

The steps to resolve this issue are, for each connector:
1. Define new schemas in openapi.json: InvokeRequestBaseV1 and DeployRequestBaseV1 with the necessary parameters to call the internal method to run deploy or invoke, common to both endpoints, with or without keychain.
2. Instantiate a new object of these types (whichever is applicable) in the endpoint implementation before call the common function with values from request and generated by connector.
3. Call the internal common method with this new object as argument.

```ts

async runInvoke(req: InvokeRequestBaseV1): Promise<InvokeContractV1Response> {
  ...
}

async runDeploy(req: DeployRequestBaseV1): Promise<DeployContractV1Response> {
  ...
}

```

**Comment by petermetz** [comment]
Yep, totally my fault for not clarifying better earlier, sorry again!

**Comment by RafaelAPB** [comment]
Until there is an explicit need for this feature, this issue is closed.
